packages/core/src/components/graphs/area-stacked.ts

```diff
@@ -0,0 +1,186 @@
+// Internal Imports
+import { Component } from "../component";
+import * as Configuration from "../../configuration";
+import { Roles, ScaleTypes, Events, TooltipTypes } from "../../interfaces";
+
+// D3 Imports
+import { area } from "d3-shape";
+import { select, color } from "d3";
+
+export class StackedArea extends Component {
+	type = "area-stacked";
+
+	areaGenerator: any;
+
+	// TODORF - Remove these listeners in destroy()
+	init() {
+		// Highlight correct area on legend item hovers
+		this.services.events.addEventListener("legend-item-onhover", e => {
+			const { hoveredElement } = e.detail;
+
+			this.parent
+				.selectAll("g.areas")
+				.transition(
+					this.services.transitions.getTransition("legend-hover-area")
+				)
+				.attr("opacity", d => {
+					if (d[0].group !== hoveredElement.datum().name) {
+						return Configuration.areas.opacity.unselected;
+					}
+
+					return Configuration.areas.opacity.selected;
+				});
+		});
+
+		// Un-highlight areas on legend item mouseouts
+		this.services.events.addEventListener("legend-item-onmouseout", e => {
+			this.parent
+				.selectAll("g.areas")
+				.transition(
+					this.services.transitions.getTransition(
+						"legend-mouseout-area"
+					)
+				)
+				.attr("opacity", Configuration.areas.opacity.selected);
+		});
+	}
+
+	addEventListeners() {
+		const self = this;
+		this.parent
+			.selectAll("path.area")
+			.on("mouseover", function(datum) {
+				const hoveredElement = select(this);
+				hoveredElement.classed("hovered", true);
+				hoveredElement
+					.transition(
+						self.services.transitions.getTransition(
+							"graph_element_mouseover_fill_update"
+						)
+					)
+					.attr(
+						"fill",
+						color(hoveredElement.attr("fill"))
+							.darker(0.2)
+							.toString()
+					);
```

Comment:
could we collapse some of these chained commands into one liners (as above comments). Some might be auto formatted by your code editor...

for example transition is mostly a one liner in the rest of the codebase
`					.transition(self.services.transitions.getTransition("graph_element_mouseover_fill_update"))`